### ISSUE TITLE
OneOf for sealed classes imported from another module

### ISSUE THREAD

**Issue by f1qwase** [javalin/javalin-openapi#235]
Hi @dzikoysk thank you for the great support of `@OneOf` for sealed classes. Unfortunately, recently I faced a new issue connected to it.

Previously I had only one submodule requiring OpenAPI spec but recently I got another one. And this second one uses some API classes from the first one. And there I discovered that automatic filling and mapping of `@OneOf` is not working for sealed classes imported from another module.

After looking into the annotation processor code I realized that it happens because the processor searches for candidate classes for `oneOf` list – (subtypes of the given sealed class) among the list of classes annotated with `@DiscrimnatorMappingName` which doesn't include any classes from an external module.

I couldn't find any quick solution to this problem and have to duplicate such classes in every module that uses them.
Do you have any idea of how to make `@OneOf` work properly (the same way as it works in its own module)? I'm ready to try to implement it if you have one.

**Comment by dzikoysk** [comment]
It's a little bit tricky, because annotation processing is supposed to work only for current sources - other modules behave quite the same as external dependencies, these sources were already processed and compiled before, so there's no point to repeat that process over and over again. 

I don't quite remember the limitations of AP, but assuming we can get an annotation from an external class, maybe we could just extend `OneOf` annotation, so there's a possibility to explicitly list all mappings. So instead of:

```
    @OneOf(
        discriminator = Discriminator(
            property = DiscriminatorProperty(
                name = "type",
                type = String::class,
                injectInMappings = true
            )
        )
    )
    sealed interface Union

    @DiscriminatorMappingName("class-a")
    data class A(val a: Int) : Union

    @DiscriminatorMappingName("class-b")
    @OpenApiName("B")
    data class C(val b: String) : Union
```

we could allow to define it like that:

```
    @OneOf(
        discriminator = Discriminator(
            property = DiscriminatorProperty(
                name = "type",
                type = String::class,
                mappings = [
                  Mapping("class-a", A::class),
                  Mapping("class-b", B::class),
                ]
            )
        )
    )
    sealed interface Union

    data class A(val a: Int) : Union

    @OpenApiName("B")
    data class C(val b: String) : Union
```

**Comment by dzikoysk** [comment]
> The only thing that worries me is that this approach assumes that oneOf values can be set in 4 different ways:
> where to have mappings [...]

I agree, let's keep it as simple as possible. I didn't pay that much attention into details, I'm completely fine with having `mappings` at root level, so directly on `@OneOf` annotation. Let's focus on that, because it should be enough to cover this case.

> mapping source priority [...]

I'd personally throw an exception for now - it seems to be simpler. In case someone will face an issue with that, we can always go back to this discussion and implement that having a real world use-case.

---

So summing up - we'll have pretty much something like that:

```kotlin
@OneOf(
    discriminator = Discriminator(
        property = DiscriminatorProperty(
            name = "type",
            type = String::class,
        )
    ),
    mappings = [
        Mapping("class-a", A::class),
        Mapping("class-b", B::class),
    ]
)
sealed interface Union
```

1. Whenever `mappings` are used, `DiscriminatorMappingName` will throw an exception to avoid duplicated mappings.
2. Whenever `mappings` are used, it replaces default `values` - in case both are used, it also throws an exception.

If it looks correct to you, feel free to open a PR 🙏
